extensions/amp-analytics/0.1/amp-analytics.js

```diff
@@ -0,0 +1,190 @@
+   * and creates a map of request name to request template. These requests can
+   * then be used while sending a request to a server.
+   */
+  generateRequests_() {
+    const requests = {};
+    for (const config of [AnalyticsConfig[this.type_], this.config_]) {
+      if (!config || !config['requests']) {
+        continue;
+      }
+      for (const name in config['requests']) {
+        if (config['requests'].hasOwnProperty(name)) {
+          requests[name] = config['requests'][name];
+        }
+      }
+    }
+    this.requests_ = requests;
+
+    // Now replace any request templates with their values.
+    const all = Object.keys(this.requests_).join('|');
+    const requestExpr = new RegExp('\{(' + all + ')\}', 'g');
+    for (const k in this.requests_) {
+      this.requests_[k] = this.expandRequest_(this.requests_[k], requestExpr,
+          this.requests_, 5);
+    }
+  }
+
+  /**
+   * Replaces template variables corresponding to a request with the value of
+   * that request. If no request is found, an empty string is inserted instead.
+   */
+  expandRequest_(formatString, expression, requests, depth) {
+    if (depth <= 0) {
+      return '';
+    }
+    const that = this;
+    depth--;
+    const request = formatString.replace(expression, function(match, name) {
+      return that.expandRequest_(requests[name], expression, requests, depth) ||
+          '';
+    });
+    return request;
+  }
+
+  /**
+   * Replaces template variables corresponding to variables with it's value
+   * provided by the trigger, the config and the  platform in that order.
+   */
+  fillVariables_(formatString) {
+
+    // TODO: Support for other variables.
+    const vars = platformVariablesFor(this.element.ownerDocument.defaultView);
+    const expr = platformVariablesExpressionFor(
+        this.element.ownerDocument.defaultView);
+    const request = formatString.replace(expr, function(match, name) {
+      let val = vars[name]();
+      if (!val && val !== 0) {
+        val = '';
+      }
+      return encodeURIComponent(val);
+    });
+    return request;
+  }
+
+  /**
+   * Callback for events that are registered by the config's triggers. This
+   * method generates the request and sends the request out.
+   */
+  handleEvent_(trigger, event) {
+    const baseConfig = AnalyticsConfig[this.type_] || {};
+    const host = this.config_['host'] || baseConfig['host'];
+    let request = this.requests_[trigger['request']];
+    if (!host || !request) {
+      return;
+    }
+
+    request = this.fillVariables_(request);
+
+    // TODO: Support for sendBeacon.
+    if (host && request) {
+      this.sendRequest('https://' + host + request);
+    }
+  }
+
+  sendRequest(request) {
+    // TODO: Logging should happen in dev mode only.
+    console./*OK*/info(this.getName() + ': Sent Request: ' + request);
+    const image = new Image();
+    image.src = request;
+    image.width = 1;
+    image.height = 1;
```

Comment:
In devmode you need to handle onload/onerror (see event-helper.js for help) and log things accordingly.